psys/daemon.py

```diff
@@ -82,3 +83,46 @@ def release_pidfile(path, fd):
         eintr_retry(os.unlink)(path)
     finally:
         eintr_retry(os.close)(fd)
+
+
+def write_pidfile(fd):
+    """Write pid to pidfile previously allocated by acquire_pidfile()"""
+
+    data = str(os.getpid())
+    datalen = len(data)
+
+    while data:
+        size = os.write(fd, data)
+        data = data[size:]
+    os.ftruncate(fd, datalen)
+
+
+def daemonize(do_fork=True, skip_fds=[]):
+    """Daemonizes current process."""
+
+    if do_fork:
+        if os.fork():
+            os._exit(0)
+        else:
+            os.setsid()
+
+            if os.fork():
+                os._exit(0)
+
+    os.chdir("/")
+    os.umask(0)
+
+    signal.signal(signal.SIGHUP, signal.SIG_IGN)
+    signal.siginterrupt(signal.SIGHUP, False)
+
+    # Redirecting stdout and stderr to the log file and
```

Comment:
Misleading comment: actually we don't redirect stdout/stderr to any log file - we redirect them to /dev/null.